src/Plugin/QueueWorker/EloquaAppCloudContentQueueWorker.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Drupal\eloqua_app_cloud\Plugin\QueueWorker;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Queue\QueueWorkerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\eloqua_rest_api\Factory\ClientFactory;
+use Drupal\Core\Queue\QueueFactory;
+use Drupal\Core\Queue\QueueInterface;
+use Psr\Log\LoggerInterface;
+
+
+/**
+ * Give it 60 seconds?
+ *
+ * @property  logger
+ * @QueueWorker(
+ *  id = "eloqua_app_cloud_content_queue_worker",
+ *  title = @Translation("The Eloqua AppCloud Queue worker for dynamic content."),
+ *  cron = {"time" = 60},
+ * )
+ */
+class EloquaAppCloudContentQueueWorker extends EloquaAppCloudQueueWorkerBase implements QueueWorkerInterface, ContainerFactoryPluginInterface {
+
+  /**
+   * @var ClientFactory
+   */
+  protected $eloquaClientFactory;
+
+  /**
+   * @var QueueFactory
+   */
+  protected $queueFactory;
+
+  /**
+   * @var  ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * @var  LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Construct.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param string $plugin_definition
+   *   The plugin implementation definition.
+   */
+  public function __construct(
+    array $configuration, $plugin_id, $plugin_definition, ClientFactory $eloqua_client_factory, QueueFactory $queueFactory, ConfigFactoryInterface $configFactory, LoggerInterface $logger) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->eloquaClientFactory = $eloqua_client_factory;
+    $this->queueFactory = $queueFactory;
+    $this->configFactory = $configFactory;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('eloqua.client_factory'),
+      $container->get('queue'),
+      $container->get('config.factory'),
+      $container->get('logger.channel.eloqua_app_cloud_queue')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($queueItem) {
```

Comment:
- [ ] ProcessItem for:
`EloquaAppCloudActionQueueWorker`
`EloquaAppCloudContentQueueWorker`
`EloquaAppCloudDecisionQueueWorker`

have a lot of duplicate code in them. Can we abstract this or move some of it over to separate protected methods?

Reply:
This would be a lot easier if the bulk API were a first class part of elomentary. 
It confuses the crap out of me when I am passing around the API object but it is not typed.
Since it is working atm I would prefer to hold off for a bit and look at that first.